src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,674 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+#
+
+# See https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md#defining-the-ci-test-arguments
+# for details about the block below.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${ALL_CLUSTERS_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: --storage-path admin_storage.json --commissioning-method on-network --discriminator 1234 --passcode 20202021 --trace-to json:${TRACE_TEST_JSON}.json --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#     factoryreset: True
+#     quiet: True
+# === END CI TEST ARGUMENTS ===
+
+
+import copy
+import inspect
+from enum import IntFlag
+
+import chip.clusters as Clusters
+from chip.clusters import ClusterObjects as ClusterObjects
+from chip.clusters.Attribute import AttributePath
+from chip.clusters.ClusterObjects import ClusterObject
+from chip.clusters.enum import MatterIntEnum
+from chip.exceptions import ChipStackError
+from chip.interaction_model import InteractionModelError, Status
+from chip.testing import global_attribute_ids
+from chip.tlv import uint
+from matter_testing_infrastructure.chip.testing.basic_composition import BasicCompositionTests
+from matter_testing_infrastructure.chip.testing.matter_testing import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_IDM_2_2(MatterBaseTest, BasicCompositionTests):
+
+    async def all_type_attributes_for_cluster(self, cluster: ClusterObjects.Cluster, desired_type: type) -> list[ClusterObjects.ClusterAttributeDescriptor]:
+        all_attributes = [attribute for attribute in cluster.Attributes.__dict__.values() if inspect.isclass(
+            attribute) and issubclass(attribute, ClusterObjects.ClusterAttributeDescriptor)]
+
+        # Hackish way to get enums to return properly -- the default behavior (under else block) returns a BLANK LIST without this workaround
```

Comment:
The comment was reworded to explain this more accurately. I left it in, just in case someone looks at the code and gets confused as to why it's there.